mvnxparser/mvnxStreamReader/mvnxStreamReader.cpp

```diff
@@ -0,0 +1,218 @@
+        string elementText = xml.text().toString().toStdString();
+        elementAttributes  = xml.attributes();
+
+        if (not xml.isCharacters()) {
+            cout << tokenString << ": " << elementName << endl;
+            if (not elementAttributes.empty()) {
+                for (auto element : elementAttributes)
+                    cout << "   " << element.name().toString().toStdString()
+                         << "=\"" << element.value().toString().toStdString()
+                         << "\"" << endl;
+            }
+        }
+
+        if (xml.isStartDocument() || xml.isEndDocument()) {
+            cout << endl;
+            continue;
+        }
+
+        // Print the content of the element
+        if (xml.isCharacters() && not xml.isWhitespace())
+            cout << tokenString << ": " << xml.text().toString().toStdString()
+                 << endl;
+    }
+}
+
+bool mvnxStreamReader::parse()
+{
+    // Initialize the XML file
+    QXmlStreamReader xml;
+    xmlFile.seek(0);
+    xml.setDevice(&xmlFile);
+
+    // Initialize the objects that will contain the metadata of the current
+    // element
+    QXmlStreamReader::TokenType tokenType;
+    string tokenString;
+    string elementName;
+    // string elementNS; // TODO: Namespace support
+    string elementText;
+    QXmlStreamAttributes elementAttributes;
+
+    // Sequentially parse the file
+    while (!xml.atEnd()) {
+        xml.readNext();
+
+        // Get the metadata of the element
+        tokenType   = xml.tokenType();
+        tokenString = xml.tokenString().toStdString();
+        elementName = xml.name().toString().toStdString();
+        // elementNS   = xml.namespaceUri().toString().toStdString();
+        elementText       = xml.text().toString().toStdString();
+        elementAttributes = xml.attributes();
+
+        // Handle the generated events
+        switch (tokenType) {
+            case QXmlStreamReader::StartDocument:
+                break;
+            case QXmlStreamReader::StartElement:
+                handleStartElement(elementName, elementAttributes);
+                break;
+            case QXmlStreamReader::Characters:
+                handleCharacters(elementText);
+                break;
+            case QXmlStreamReader::Comment:
+                handleComment(elementText);
+                break;
+            case QXmlStreamReader::EndElement:
+                handleStopElement(elementName);
+                break;
+            case QXmlStreamReader::EndDocument:
+                xmlTreeRoot = elementsLIFO.front();
+                elementsLIFO.pop_back();
+                assert(elementsLIFO.empty());
+                break;
+            default:
+                break;
+        }
+    }
+    return false;
+}
+
+attributes_t
+mvnxStreamReader::processAttributes(QXmlStreamAttributes elementAttributes)
+{
+    attributes_t attributes;
+    if (not elementAttributes.empty()) {
```

Comment:
Is there a reason why you used `not` and not `!`?
If you don't mind I prefer the `!`